### DISCUSSION TITLE
using a certificate without a private key results in a confusing error message

### DISCUSSION THREAD

**Discussion by wallaceturner** [ravendb/ravendb#14178] [General]
I'm using raven client 4.2.110 - i'm getting a certificate from the `X509Store` and assigning it to the raven `DocumentStore`

`store.Certificate = clientCertificate;`

If you use a valid certificate but one which does not have a private key the driver lets you do this and it fails later with this error message

>>This server requires client certificate for authentication, but none was provided by the client.

This is a bit confusing and could be caught earlier? e.g.

```
               if (clientCertificate.HasPrivateKey)
                {
                    store.Certificate = clientCertificate;
                }
                else
                {
                    throw new Exception("does not have private key for certificate. did you import the crt instead of the pfx?");
                }
```

**Comment by ayende** [comment]
That is already the case in the 5.x version of RavenDB, see:

https://github.com/ravendb/ravendb/blob/42c5e4241fb3952996101300e1f6153bd1163949/src/Raven.Client/Documents/DocumentStore.cs#L227

Note that 4.2 is on LTS only right now, and it will go out of support in about two months. As such, I'll recommend upgrading to 5.2 / 5.3
